tools/distrib/fix_build_deps.py

```diff
@@ -0,0 +1,410 @@
+    'src/proto/grpc/lb/v1/load_reporter.grpc.pb.h':
+        '//src/proto/grpc/lb/v1:load_reporter_proto',
+    'src/proto/grpc/lb/v1/load_balancer.upb.h':
+        'grpc_lb_upb',
+    'src/proto/grpc/reflection/v1alpha/reflection.grpc.pb.h':
+        '//src/proto/grpc/reflection/v1alpha:reflection_proto',
+    'src/proto/grpc/gcp/transport_security_common.upb.h':
+        'alts_upb',
+    'src/proto/grpc/gcp/altscontext.upb.h':
+        'alts_upb',
+    'src/proto/grpc/lookup/v1/rls.upb.h':
+        'rls_upb',
+    'src/proto/grpc/lookup/v1/rls_config.upb.h':
+        'rls_config_upb',
+    'src/proto/grpc/lookup/v1/rls_config.upbdefs.h':
+        'rls_config_upbdefs',
+    'src/proto/grpc/testing/xds/v3/csds.grpc.pb.h':
+        '//src/proto/grpc/testing/xds/v3:csds_proto',
+    'xds/data/orca/v3/orca_load_report.upb.h':
+        'xds_orca_upb',
+    'xds/service/orca/v3/orca.upb.h':
+        'xds_orca_service_upb',
+    'xds/type/v3/typed_struct.upb.h':
+        'xds_type_upb',
+}
+
+SKIP_DEPS = {'google/api/expr/v1alpha1/syntax.upb.h'}
+
+
+class FakeSelects:
+
+    def config_setting_group(self, **kwargs):
+        pass
+
+
+def grpc_cc_library(name,
+                    hdrs=[],
+                    public_hdrs=[],
+                    srcs=[],
+                    select_deps=None,
+                    tags=[],
+                    **kwargs):
+    if select_deps or 'nofixdeps' in tags or 'grpc-autodeps' not in tags:
+        no_update.add(name)
+    score = len(public_hdrs + hdrs)
+    if 'avoid_dep' in tags:
+        score += 1000000
+    if 'nofixdeps' in tags:
+        score += 1000
+    for hdr in hdrs + public_hdrs:
+        vendors.setdefault(hdr, []).append(Vendor(name, score))
+    inc = set()
+    for src in hdrs + public_hdrs + srcs:
+        for line in open(src):
+            m = re.search(r'#include <(.*)>', line)
+            if m:
+                inc.add(m.group(1))
+            m = re.search(r'#include "(.*)"', line)
+            if m:
+                inc.add(m.group(1))
+            if 'grpc::g_glip' in line or 'grpc:g_core_codegen_interface' in line:
+                needs_codegen_base_src.add(name)
+    consumes[name] = list(inc)
+
+
+def choose_vendor(library, vendors):
+    best = None
+    for vendor in vendors:
+        if vendor.name == library:
+            return None
+        if best is None or vendor.score < best.score:
+            best = vendor
+    return best.name
+
+
+def buildozer(cmd, target):
+    buildozer_commands.append('%s|%s' % (cmd, target))
+
+
+def buildozer_set_list(name, values, target, via=""):
+    if not values:
+        buildozer('remove %s' % name, target)
+        return
+    adjust = via if via else name
+    buildozer('set %s %s' % (adjust, ' '.join('"%s"' % s for s in values)),
+              target)
+    if via:
+        buildozer('remove %s' % name, target)
+        buildozer('rename %s %s' % (via, name), target)
+
+
+exec(
+    open('BUILD', 'r').read(), {
```

Comment:
This should be run as part of the sanity tests, right? If not, I'm imagining a situation where we use a new function in the BUILD file and end up breaking this script without realizing it.

Reply:
It gets run as part of generate_projects, which gets run as part of the sanity tests, so I think we're ok for now.